main.go

```diff
@@ -117,6 +137,11 @@ func realMain() int {
 		fmt.Fprintf(io.Err(), "Error executing hcp: %s\n", err.Error())
 	}
 
+	// Cancel the update check if it hasn't finished already
+	if checker.UpdateToDisplay() && !isAutocomplete() {
```

Comment:
Would it make sense to remove the `UpdateToDisplay` function and just rely on `Display` to be a noop if there is nothing to display?